### DISCUSSION TITLE
`github.dev` stuck on "Setting up your web editor"

### DISCUSSION THREAD

**Discussion by Errorbot1122** [community/community#15694] [Other Feature Feedback, Questions, & Ideas]
For some reason `github.dev` gets stuck on the "Setting up" screen indefinitely.

![image](https://user-images.githubusercontent.com/65428515/165589238-38b1b62d-e107-4a69-b49b-38c1f30263a6.png)

I'm not sure why, but I think it may have to do with a problem with `vscode.dev`.

![image](https://user-images.githubusercontent.com/65428515/165589430-5a38c3c1-8c02-4e8e-829c-afc21642cf9a.png)

I'm not sure if this is a bug or something else, but if you need to know, I can send you my system specs, just ask!

*BTY: I'm using a **Chromebook**, and I currently do not have **codespaces**, I just think that this is the most relevant category for this problem.*

**Comment by KevinMSampson** [comment]
I'm having the same issue on multiple browsers across mac & windows.
Looks like the Content Security Policy is blocking some new directives from different vscode urls

**Reply by ia-mfriegang** [reply]
I was getting the same error and was able to fix it by manually creating a code space, instructions are under the "Browser cannot connect" section of this article https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-your-connection-to-codespaces

**Reply by KevinMSampson** [reply]
This isn't specific to Codespaces, I'm using GitHub.dev (browser-only client-side vs code)

**Reply by ia-mfriegang** [reply]
Yeah I just realized that :/ same issue still here. Are you in a corporate environment? I have a feeling its some security policy somewhere

**Reply by KevinMSampson** [reply]
That I am 👍🏼 
I did get past this screen by having our networking team troubleshoot and add some urls to the whitelist.
It doesn't help that all the GitHub whitelist IPs in their docs are IP-specific instead of domain

**Reply by ia-mfriegang** [reply]
just confirmed it was the same issue on my end. Its working as expected now

**Reply by krishkumarsingh1545** [reply]
> That I am 👍🏼 
> I did get past this screen by having our networking team troubleshoot and add some urls to the whitelist.
> It doesn't help that all the GitHub whitelist IPs in their docs are IP-specific instead of domain

Tell me Sir how you passed that page.
Mine is just stucked there.

**Comment by jfullstackdev** [comment]
never experienced that on Chrome, I even use Codespaces which is heavier than that 

maybe low memory, browser extension, browser itself, firewall, antivirus local setup

**Reply by krishkumarsingh1545** [reply]
Oh Whatever I fixed it. 
The problem was the certificate was expired so I updated it manually

**Comment by adityawisecoder** [comment]
I also encountered the same problem, but I was able to resolve it by **enabling my VPN** and accessing the web-editor, which appeared to be working fine. Once I **disabled my VPN**, the issue was resolved and everything returned to normal. 

Hopefully, this solution will work for you too!✌️

**Reply by krishkumarsingh1545** [reply]
My SSL Certificate was expired.
So I updated it with a new one.
That worked for me and never had that problem again. 
I think using a VPN might modified your certificates too or something like that. 
By the way, thanks for the help.

**Reply by rhoitjadhav** [reply]
Not sure how, but enabling VPN has magically worked.

**Comment by tajamulkhann** [comment]
The best solution is to Connect to a VPN. The issue will be resolved 100%

Need a Free VPN Connection? https://protonvpn.com/

**Reply by yungjakey** [reply]
how in the world is this a proper place to drop an advertisement?

**Comment by pablitok** [comment]
Hi I'm having the same issue with multliple browsers:
- Safari Version 18.1 (19619.2.8.111.5, 19619)
- Chrome Beta 132.0.6834.15
- Chrome Stable 131.0.6778.86

All in Mac 14.7.1 (23H222) with M1

Tried with my work vpn and without it, I cannot use other vpn. Also I'm seeing the same issue with other coworkers

**Reply by krishkumarsingh1545** [reply]
So, My SSL Certificate was expired.
So I updated it with a new one.
That worked for me and never had that problem again.
I think using a VPN might modified your certificates too or something like that.
Try to look if any certificate file is expired and update it.

**Reply by pablitok** [reply]
Didn't have luck with a vpn but now it's working without any change

**Comment by sriA-97** [comment]
I downloaded the latest version of Git and that worked for me, hope it works for you as well.

**Reply by sriA-97** [reply]
try updating the git to the latest version in your local machine.

**Comment by shiva-srivastav** [comment]
I also face same issue.
![image](https://github.com/user-attachments/assets/6c5b44fe-14c3-44d4-a819-befe288bcfe8)

**Reply by Debjyoti30** [reply]
I am also facing the same issue. Please do share the drill if this issue has been resolved for you.

**Reply by sriA-97** [reply]
try updating the git to the latest version in your local machine.

**Reply by Debjyoti30** [reply]
It started working just today without any changes being made. It might be related to quotas, but I am not aware of the specifics.

**Reply by shiva-srivastav** [reply]
now its working.

**Comment by snapfast** [comment]
Refused to connect to 'https://main.vscode-cdn.net/sourcemaps/e3a5acf…/core/vs/workbench/workbench.web.main.internal.js.map' because it violates the following Content Security Policy directive: "connect-src 'self' https://assets.github.dev/ https://*.rel.tunnels.api.visualstudio.com wss://*.tunnels.api.visualstudio.com/ http://*.dev.core.vsengsaas.visualstudio.com https://*.gallerycdn.vsassets.io https://vortex.data.microsoft.com/collect/v1 https://vscode-sync.trafficmanager.net/ https://vscode-sync-insiders.trafficmanager.net/ https://vscode-sync-test.trafficmanager.net/ https://marketplace.visualstudio.com/_apis/public/gallery/ https://az764295.vo.msecnd.net/experiments/vscode-experiments.json https://vscodeexperiments.azureedge.net/experiments/vscode-experiments.json https://vscodeweb.azureedge.net/insider/ https://vscodeweb.azureedge.net/stable/ https://main.vscode-cdn.net/core/ https://main.vscode-cdn.net/extensions/ https://main.vscode-cdn.net/insider/ https://main.vscode-cdn.net/stable/ https://default.exp-tas.com/vscode/ab https://code.visualstudio.com/raw/ https://vscode-auth.github.com https://*.gallery.vsassets.io/_apis/public/gallery/ https://*.vscode-unpkg.net/ https://github-cloud.githubusercontent.com/alambic/media/ https://bingsettingssearch.trafficmanager.net/ https://vscode.search.windows.net/ https://vscode.blob.core.windows.net/gallery/index/ https://marketplace.visualstudio.com/items/ https://az764295.vo.msecnd.net/extensions/marketplace.json https://az764295.vo.msecnd.net/extensions/workspaceRecommendations.json.gz https://*.review-lab.github.com https://update.code.visualstudio.com/api/ https://mobile.events.data.microsoft.com/ https://az764295.vo.msecnd.net/profile-templates/ https://vscode.download.prss.microsoft.com/ https://api.github.com/".



Facing this issue

**Comment by tscpiece** [comment]
I couldn't connect to github.dev (codespace) or vscode.dev on my home network, even after clearing cookies and turning off the extensions. 
I could ssh into it by `gh cs ssh` with no issues.
Then I connected to a VPN and now the website's working.

**Comment by Stan370** [comment]
This is a known bug with `github.dev` (the VS Code web editor) — it can get stuck forever on **“Setting up your web editor”** when “Direct Mode” (the browser-based VS Code runtime) fails to initialize properly.
it’s often caused by cached state, service worker corruption, or network/extension conflicts.

`github.dev` can open in two ways:

* **Direct Mode (default)** → VS Code runs *entirely in browser*, using IndexedDB and web containers.
* **Server Mode (via vscode.dev proxy)** → VS Code runs on Microsoft’s cloud backend and syncs via websocket.

if Direct Mode breaks, you’ll see that “Setting up your web editor…” message forever.

---


open Chrome devtools on github.dev:

```bash
Ctrl+Shift+I (Windows/Linux)
Cmd+Option+I (Mac)
```

then:

**Application → Service Workers → Unregister**
and clear:

```
Application → Clear storage → "Clear site data"
```

then reload:

```
https://github.dev/<user>/<repo>
```

---

or force disable Direct Mode via localStorage

**Comment by Darginec05** [comment]
same issue in every browser

**Reply by krishkumarsingh1545** [reply]
yeah changing browser won't fix the issue, you have to download the updated version of expired SSL certificate or just try any VPN

**Comment by Ninad7n** [comment]
This fixed the issue for me
Network → Wi-Fi → DNS 
8.8.8.8
8.8.4.4

**Reply by ShaileshParmar11** [reply]
I worked for me

**Comment by DevZ44d** [comment]
I also encountered the same problem, but I was able to resolve it by enabling my VPN and accessing the web-editor, which appeared to be working fine. Once I disabled my VPN, the issue was resolved and everything returned to normal.

Hopefully, this solution will work for you too!

**Comment by PizzaHut1212** [comment]
If GitHub.dev is stuck on “Setting up your web editor,” try these simple steps: use a modern browser, disable interfering extensions, clear your cache, and reload. Logging out and back in or opening a smaller repo can also help. Check your internet connection and GitHub’s status page. These usually fix the problem quickly and you can use also this vpn to solve your issu [kalkulator](https://kalkulatoronline.de/)
